src/api/config/config.go

```diff
@@ -0,0 +1,69 @@
+package config
+
+import (
+	"os"
+
+	"github.com/imdario/mergo"
+)
+
+// ConfigurationWithOverrides includes default Configuration values
+// and environment specific ones
+type configurationWithOverrides map[string]Configuration
+
+// Configuration is the the resulting environment based Configuration
+// For now it only includes Cors info
+type Configuration struct {
+	Cors
+}
+
+// Cors configuration used during middleware setup
+type Cors struct {
+	AllowedOrigins []string
+	AllowedHeaders []string
+}
+
+func currentEnvironment() string {
+	env := os.Getenv("ENVIRONMENT")
+	if env == "" {
+		env = "production"
+	}
+	return env
+}
+
+func readConfigWithOverrides() configurationWithOverrides {
```

Comment:
Instead of having a yaml file, we just have the structs here.

That removes the requirement of binding the yaml file to the binary file.